src/pages/index.js

```diff
@@ -107,18 +131,80 @@ profileAddButton.addEventListener("click", function () {
 
 /* Functions */
 
-// Define the createCard function
-function createCard(cardData) {
-  // Select the template element and clone its contents
-  const cardInfo = new Card(cardData, "#card__template", function () {
-    openZoomPicture(cardData.link, cardData.name);
+function handleAvatarUpdate({ link }) {
+  return api.updateAvatar({ avatar: link }).then((res) => {
+    userInfo.setUserInfo({
+      title: res.name,
+      description: res.about,
+      avatar: res.avatar,
+    });
   });
+}
+
+function cardLikeClick(cardId, isLiked) {
+  if (isLiked) {
+    return api.likeCard(cardId).catch((error) => {
+      console.error(`Failed to like card with ID ${cardId}:`, error);
+    });
+  } else {
+    return api.unlikeCard(cardId).catch((error) => {
+      console.error(`Failed to unlike card with ID ${cardId}:`, error);
+    });
+  }
+}
+
+function createCard(cardData) {
+  const cardInfo = new Card(
+    cardData,
+    "#card__template",
+    function () {
+      openZoomPicture(cardData.link, cardData.name);
+    },
+    handleDeleteCard,
+    cardLikeClick
+  );
   const cardElement = cardInfo.getView();
   return cardElement;
 }
+function handleDeleteCard(card, cardElement) {
+  popupWithConfirm.open();
+  console.log("ghdd", card);
+  popupWithConfirm.setSubmitAction(() => {
+    return api
+      .deleteCard(card._id)
+      .then(() => {
+        cardElement.remove();
+
+        popupWithConfirm.close();
+      })
+      .catch((err) => {
+        console.err(`Error deleting card: ${err}`);
+      });
+  });
+}
 
 function openZoomPicture(src, alt) {
   const data = { name: alt, link: src };
 
   popupWithImage.open(data);
 }
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "93b64164-7992-45be-bad2-1f9098f0ec32",
+    "Content-Type": "application/json",
+  },
+});
+
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo({
+    title: userData.name,
+    description: userData.about,
+    avatar: userData.avatar,
+  });
+});
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**